src/command_modules/azure-cli-network/azure/cli/command_modules/network/dns_zone.py

```diff
@@ -22,3 +22,21 @@ def create_dns_zone(client, resource_group_name, zone_name, location='global', t
         kwargs['if_none_match'] = '*'
 
     return client.create_or_update(**kwargs)
+
+
+#pylint: disable=too-many-arguments
+def update_dns_zone(client, resource_group_name, zone_name, location='global', if_match=None,
+                    **operation_config):
+    body = {'location': location}
+
+    kwargs = {
+        'resource_group_name':resource_group_name,
+        'zone_name': zone_name,
+        'parameters': body,
+        'operation_config': operation_config
+    }
```

Comment:
This breaks the logical flow of a generic update. Updating the parameters should happen in the method `cf_dns_mgmt_zones` (and probably already does). This happens before the generic stuff is applied because the generic flags can and will override convenience flags by design. By putting this in the setter, you would be overriding what is set by the generic flags.